File: temporal-python-workers/start_hello.py
#!/usr/bin/env python3
"""
Starter script — kicks off a HelloWorldWorkflow execution.

Usage:
    python start_hello.py [NAME]
"""

import asyncio
import sys

from temporalio.client import Client

from config import settings
from workers.hello_world.workflows import HelloWorldWorkflow


async def main() -> None:
    name = sys.argv[1] if len(sys.argv) > 1 else "World"

    client = await Client.connect(
        settings.server_url,
        namespace=settings.namespace,
    )

    print(f"Starting HelloWorldWorkflow for '{name}' …")

    handle = await client.start_workflow(
        HelloWorldWorkflow.run,
        name,
        id=f"hello-world-{name.lower().replace(' ', '-')}",
        task_queue="hello-world-python-queue",
    )

    print(f"Workflow started: id={handle.id}, run_id={handle.result_run_id}")
    print("Waiting for result (this is a long-running task) …")

    result = await handle.result()
    print(f"Result: {result}")


if __name__ == "__main__":
    asyncio.run(main())

File: temporal-workers-java/src/main/java/com/temporal/workers/StartHello.java
package com.temporal.workers;

import com.temporal.workers.config.WorkerConfig;
import com.temporal.workers.helloworld.HelloResult;
import com.temporal.workers.helloworld.HelloWorldWorkflow;
import io.temporal.client.WorkflowClient;
import io.temporal.client.WorkflowClientOptions;
import io.temporal.client.WorkflowOptions;
import io.temporal.serviceclient.WorkflowServiceStubs;
import io.temporal.serviceclient.WorkflowServiceStubsOptions;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * CLI starter — kicks off a HelloWorldWorkflow execution.
 *
 * Usage:
 *     java -cp app.jar com.temporal.workers.StartHello [NAME]
 */
public class StartHello {

    private static final Logger logger = LoggerFactory.getLogger(StartHello.class);

    public static void main(String[] args) {
        String name = (args.length > 0) ? args[0] : "World";

        WorkerConfig config = WorkerConfig.getInstance();

        WorkflowServiceStubs service = WorkflowServiceStubs.newServiceStubs(
                WorkflowServiceStubsOptions.newBuilder()
                        .setTarget(config.getServerUrl())
                        .build()
        );

        WorkflowClient client = WorkflowClient.newInstance(
                service,
                WorkflowClientOptions.newBuilder()
                        .setNamespace(config.getNamespace())
                        .build()
        );

        String workflowId = "hello-world-" + name.toLowerCase().replace(' ', '-');

        HelloWorldWorkflow workflow = client.newWorkflowStub(
                HelloWorldWorkflow.class,
                WorkflowOptions.newBuilder()
                        .setWorkflowId(workflowId)
                        .setTaskQueue("hello-world-queue")
                        .build()
        );

        logger.info("Starting HelloWorldWorkflow for '{}' …", name);

        HelloResult result = workflow.run(name);

        logger.info("Result: {}", result);
        System.out.printf("✅ %s%n", result.getGreeting());

        System.exit(0);
    }
}
